### PR TITLE
feat(exitcode): Add exit code sugar method

### PR DESCRIPTION
**Pull request by bearrito**
## What does this PR do?
Add syntax sugar on exit code handling.

## Why is it important?
I don't know. I was just looking at `good first issue` tags
https://github.com/testcontainers/testcontainers-go/issues/2205

## How to test this PR
Run the added unit tests.

### CODE REVIEW

File: wait/exec.go [hunk 1]
```diff
@@ -45,6 +45,13 @@ func (ws *ExecStrategy) WithStartupTimeout(startupTimeout time.Duration) *ExecSt
 	return ws
 }
 
+func (ws *ExecStrategy) WithExitCode(exitCode int) *ExecStrategy {
+
```
Comment (mdelapenya) [inline wait/exec.go:49]: Lint will probably complain
```suggestion
```

**Review by mdelapenya — COMMENTED** [review]
LGTM! I'll let the CI to run once the lint is fixed, and if passes, will approve it.

Thanks for your time!

File: wait/exec.go (hunk 1 above)
Comment (bearrito) [reply wait/exec.go]: Fixed. Is there make command or pre-commit we should be running that will do what CI is doing?

Comment (mdelapenya) [reply wait/exec.go]: There is no pre-commit (https://pre-commit.com/) at the developer side, but the CI will use golangci-lint. You can run it with `make lint` from the root dir of the project.

File: wait/exec_test.go [hunk 2]
```diff
@@ -156,6 +156,27 @@ func TestExecStrategyWaitUntilReady_CustomExitCode(t *testing.T) {
 	}
 }
 
+func TestExecStrategyWaitUntilReadyWithSugaryExitCode(t *testing.T) {
```
Comment (mdelapenya) [inline wait/exec_test.go:159]: Not a blocker, but more consistent name with other test functions
```suggestion
func TestExecStrategyWaitUntilReady_ExitCode(t *testing.T) {
```

Comment (bearrito) [reply wait/exec_test.go]: @mdelapenya Updated.

**Comment by bearrito** [comment]
Merged in main

File: wait/exec_test.go [hunk 3]
```diff
@@ -156,6 +156,27 @@ func TestExecStrategyWaitUntilReady_CustomExitCode(t *testing.T) {
 	}
 }
 
+func TestExecStrategyWaitUntilReadyWithSugaryExitCode_MatchesAndTimeout(t *testing.T) {
```
Comment (mdelapenya) [inline wait/exec_test.go:159]:
```suggestion
func TestExecStrategyWaitUntilReady_withExitCode(t *testing.T) {
```

**Review by mdelapenya — COMMENTED** [review]
Hi @bearrito I think once we change the test method, this is ready to be merged.

Thanks for your time contributing to the library!

**Comment by bearrito** [comment]
@mdelapenya  Updated.

**Pull request merged**